### PR TITLE
Fade out matched blocks

### PR DESCRIPTION
**Pull request by dtreiter**
Previously when blocks were matched the would be removed instantly.
Now they have an animated fade out. NOTE: This change now uses the
previously unused BlockState GONE.

### CODE REVIEW

File: app/scripts/Grid.js [hunk 1]
```diff
@@ -306,6 +313,7 @@
       // most ticks, there will be nothing to do, so
       // this is pretty wasteful
       makeBlocksFall();
+      removeFadedBlocks();
```
Comment (ymmn) [inline app/scripts/Grid.js]: use a promise to make this really pretty. look at how i did swapping.

it should look like:

```
matchedBlock.die().then(function(block) { 
  removeBlock(block);
}
```

Comment (ymmn) [inline app/scripts/Grid.js]: the entire thing should really be event driven. anywhere we loop through all the blocks (except to call `tick`) is probably bad and could be done in a much better way

**Comment by dtreiter** [comment]
Refactored to use promises. I have to admit it's pretty nice.

File: app/scripts/Grid.js [hunk 2]
```diff
@@ -124,7 +124,7 @@
         }
       }
     };
-
+
```
Comment (ymmn) [inline app/scripts/Grid.js:127]: you should always run your code thru jshint. it will catch things like trailing whitespace

**Comment by ymmn** [comment]
lgtm

**Pull request merged**